cloud/scope/machinepool.go

```diff
@@ -41,37 +39,35 @@ import (
 type (
 	// MachinePoolScopeParams defines the input parameters used to create a new MachinePoolScope.
 	MachinePoolScopeParams struct {
-		AzureClients
 		Client           client.Client
 		Logger           logr.Logger
 		MachinePool      *capiv1exp.MachinePool
 		AzureMachinePool *infrav1exp.AzureMachinePool
-		ClusterScope     *ClusterScope
+		ClusterScope     ClusterScope
 	}
 
 	// MachinePoolScope defines a scope defined around a machine pool and its cluster.
 	MachinePoolScope struct {
 		logr.Logger
-		AzureClients
 		client           client.Client
 		patchHelper      *patch.Helper
 		MachinePool      *capiv1exp.MachinePool
 		AzureMachinePool *infrav1exp.AzureMachinePool
-		ClusterScope     azure.ClusterDescriber
+		ClusterScope
```

Comment:
I might keep the interface for now, unless ClusterScope is already very small?

I thought this was a clever way to incrementally migrate us off of using the whole scope object, while starting to constrain an exposed interface. I'd assume it would merge with ClusterScope when the per-service bits get abstracted out so cluster scope is only shared vars (resourcegroup, name, location).

Reply:
Lemme know what you think of this new diff (I think it's better 😃 ) 

I kept all the commits to be able to revert if needed, will squash once we have consensus